metaflow/exception.py

```diff
@@ -160,6 +160,15 @@ def __init__(self, msg, unhandled):
         self.artifact_names = unhandled
 
 
+class ServiceException(MetaflowException):
```

Comment:
why move this here? - this is tied to the service implementation and ideally shouldn't be pulled into the core